### DISCUSSION TITLE
How does tiddlyweb and localstorage plugin handle deletion

### DISCUSSION THREAD

**Discussion by linonetwo** [TiddlyWiki/TiddlyWiki5#7527] [General]
When fixing https://github.com/tiddly-gittly/tw-mobile-sync/issues/7 , I think I may need to know how other plugins handle async deletion.

If a client is offline, and later online, how will it let tiddlyweb server know there is an deletion? By sending a full tiddlers list to it? (This will be huge, I have 20k tiddlers now).

I think it may just record the recent changes since last sync, and it contains changedTiddlers, which has delete and modify field set to `true`.

I need to read the code to learn this. Just record here.

**Comment by Jermolene** [comment]
Hi @linonetwo 

The tiddlyweb adaptor handles deletions via the syncer module which keeps track of the tiddlers that are known on the server; if it later finds a tiddler that it marked as present on the server but not present on the client, then it will send a deletion request to the server. The code is here:

https://github.com/Jermolene/TiddlyWiki5/blob/cce23ac6cddbccc88a848dcc5c456e57c01b2c20/core/modules/syncer.js#L566-L575

The browser storage plugin uses similar logic in https://github.com/Jermolene/TiddlyWiki5/blob/master/plugins/tiddlywiki/browser-storage/rawmarkup.js

**Reply by linonetwo** [reply]
Thank you. I see, this task queue is in-memory, so if network broken, and a user restart wiki before the network is re-connected (before `syncadaptor.isReady` become true), the delete state will be lost. Next time when network is connected again, that user will find deleted tiddlers "rewind".

So I can consider this problem is not resolved in tiddlyweb too.

(Looks like browser storage plugin won't have this disconnected problem...)

But maybe we can utilize [ Developer demo of alternate wiki store implementation #7329 ](https://github.com/Jermolene/TiddlyWiki5/pull/7329) to also store tasks in sqlite...Or just use

**Reply by Jermolene** [reply]
Hi @linonetwo in v5.2.7 we extended the browser-storage plugin to work with the tiddlyweb plugin to provide support for broken network connections.
